src/main/java/hudson/plugins/cobertura/CoberturaPublisher.java

```diff
@@ -43,64 +46,46 @@
 
 import org.apache.commons.beanutils.ConvertUtils;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.StaplerRequest;
 
 /**
  * Cobertura {@link Publisher}.
  *
  * @author Stephen Connolly
  */
-public class CoberturaPublisher extends Recorder {
+public class CoberturaPublisher extends Recorder implements SimpleBuildStep {
+
+    private String coberturaReportFile;
 
-    private final String coberturaReportFile;
+    private boolean onlyStable;
 
-    private final boolean onlyStable;
+    private boolean failUnhealthy;
 
-    private final boolean failUnhealthy;
+    private boolean failUnstable;
 
-    private final boolean failUnstable;
+    private boolean autoUpdateHealth;
 
-    private final boolean autoUpdateHealth;
+    private boolean autoUpdateStability;
 
-    private final boolean autoUpdateStability;
+    private boolean zoomCoverageChart;
+
+    private int maxNumberOfBuilds = 0;
 
-    private final boolean zoomCoverageChart;
-    
-    private final int maxNumberOfBuilds;
-    
     private boolean failNoReports = true;
 
-    private CoverageTarget healthyTarget;
+    private CoverageTarget healthyTarget = new CoverageTarget();
 
-    private CoverageTarget unhealthyTarget;
+    private CoverageTarget unhealthyTarget = new CoverageTarget();
 
-    private CoverageTarget failingTarget;
+    private CoverageTarget failingTarget = new CoverageTarget();
 
     public static final CoberturaReportFilenameFilter COBERTURA_FILENAME_FILTER = new CoberturaReportFilenameFilter();
 
-    private final SourceEncoding sourceEncoding;
+    private SourceEncoding sourceEncoding = SourceEncoding.UTF_8;
 
-    /**
-     * @param coberturaReportFile the report directory
-     * @stapler-constructor
-     */
     @DataBoundConstructor
-    public CoberturaPublisher(String coberturaReportFile, boolean onlyStable, boolean failUnhealthy, boolean failUnstable, 
```

Comment:
It's redundant and might not cause issues when we release, but provides backwards compatibility for others. By this I mean if there's another plugin (custom (not public) plugin, library, or another plugin in jenkinsci) which uses this constructor, then it would most likely fail to load, or they would not need to re-build their plugin to be compatible with newer versions. (see [here](https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility) for more information).

The best way to resolve this issue is:

1. Remove the `@DataBoundConstructor` from the old Constructor.
2. Set it to `@Deprecated` (so others know not to use it in the future).
3. Put the new `@DataBoundConstructor` around the new constructor.
4. Set `@DataBoundSetter` annotations around all setter methods (from the original constructor).

The link @jglick posted after the first paragraph states:

```
For Java-level compatibility, leave any previous constructors in place, but mark them @Deprecated. Also remove @DataBoundConstructor from them (there can be only one).
```